lib/Indexer/Model/Indexer.php

```diff
@@ -15,17 +16,33 @@ class Indexer
 
     private DirtyDocumentTracker $dirtyDocumentTracker;
 
-    public function __construct(IndexBuilder $builder, Index $index, FileListProvider $provider, ?DirtyDocumentTracker $dirtyDocumentTracker = null)
-    {
+    private FileListProvider $stubIndexProvider;
+
+    public function __construct(
+        IndexBuilder $builder,
+        Index $index,
+        FileListProvider $provider,
+        FileListProvider $stubIndexProvider,
+        ?DirtyDocumentTracker $dirtyDocumentTracker = null
+    ) {
         $this->builder = $builder;
         $this->index = $index;
         $this->provider = $provider;
+        $this->stubIndexProvider = $stubIndexProvider;
         $this->dirtyDocumentTracker = $dirtyDocumentTracker ?: new NullDirtyDocumentTracker();
     }
 
-    public function getJob(?string $subPath = null): IndexJob
+    /**
+     * @return Generator<IndexJob>
+     */
+    public function getJobs(?string $subPath = null): Generator
     {
-        return new IndexJob(
+        yield new IndexJob(
+            $this->builder,
+            $this->stubIndexProvider->provideFileList($this->index, $subPath)
+        );
+
+        yield new IndexJob(
             $this->builder,
             $this->provider->provideFileList($this->index, $subPath)
         );
```

Comment:
ideally the Indexer should not have direct knowledge of these two sources, but maybe it's good enough for now.